tracing/src/main/java/com/palantir/tracing/DetachedSpan.java

```diff
@@ -69,7 +70,33 @@ static DetachedSpan start(
      * instead of thread state.
      */
     @MustBeClosed
-    CloseableSpan childSpan(String operationName, SpanType type);
+    default CloseableSpan childSpan(String operationName, SpanType type) {
+        return childSpan(operationName, NoTagRecorder.INSTANCE, NoTagRecorder.INSTANCE, type);
+    }
+
+    /**
+     * Equivalent to {@link #childSpan(String, Map, SpanType)} using a {@link SpanType#LOCAL span}.
+     */
+    @MustBeClosed
+    default CloseableSpan childSpan(String operationName, Map<String, String> metadata) {
+        return childSpan(operationName, metadata, SpanType.LOCAL);
+    }
+
+    @MustBeClosed
+    default <T> CloseableSpan childSpan(String operationName, TagRecorder<? super T> recorder, T data) {
+        return childSpan(operationName, recorder, data, SpanType.LOCAL);
+    }
+
+    /**
+     * Equivalent to {@link #childSpan(String, SpanType)}, but using {@link Map metadata} tags.
+     */
+    @MustBeClosed
+    default CloseableSpan childSpan(String operationName, Map<String, String> metadata, SpanType type) {
+        return childSpan(operationName, MapTagRecorder.INSTANCE, metadata, type);
+    }
+
+    @MustBeClosed
+    <T> CloseableSpan childSpan(String operationName, TagRecorder<? super T> recorder, T data, SpanType type);
```

Comment:
Note I did not update the `completeAndStartChild` methods defined below (not visible in the default GH diff) because it's not obvious that the `operationName` argument would be associated with the child trace, while tags would be associated with its parent.

It's easy enough to implement that logic as needed, and produces more obvious code:
```java
 try (CloseableSpan child = detached.childSpan(operationName, type)) {
    detached.complete(<tags>);
    // work    
}
 ```